### PR TITLE
DataUp-181 Prevent Reaper from quitting

### PR DESCRIPTION
**Pull request by bio-boris**
# Description of PR purpose/changes

Prevent the job reaper from exiting on error, just continue trying to reap

# Testing Instructions
Didn't write tests for this script.

# Dev Checklist:

- [ ] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have run Black and Flake8 on changed Python Code manually or with git precommit (and the travis build passes)

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

**Comment by bio-boris** [comment]
Should try catch the slack client call as well!!

File: bin/PurgeHeldJobs.py [hunk 1]
```diff
@@ -173,4 +173,4 @@ def handle_hold_event(event):
             time.sleep(5)
         except Exception as e:
             slack_client.ee2_reaper_failure(endpoint=ee2_endpoint)
-            sys.exit(f"{e}")
```
Comment (leia-sefkin) [inline bin/PurgeHeldJobs.py:176]: is there any concern over keeping the reaper running when there's a legitimate error here?

**Review by leia-sefkin — COMMENTED** [review]
I'm a bit out of context here so not sure my review can be all that helpful. Seems like an ok thing to try, but I'm not sure of the consequences of not quitting when we do encounter and exception here. Other concerns might be whether slack is the best location to be recording these errors - if that's a reliable process.

**Review by eamahanna — COMMENTED** [review]
My thoughts largely align with Leia's. I'm not sure the reason and consequences for this change?

File: bin/PurgeHeldJobs.py (hunk 1 above)
Comment (bio-boris) [reply bin/PurgeHeldJobs.py:176]: No, the reaper should never end

**Comment by bio-boris** [comment]
> My thoughts largely align with Leia's. I'm not sure the reason and consequences for this change?

The reason is because I saw the reaper had exited, when it should have kept running

**Review by Tianhao-Gu — APPROVED** [review]
👍 GLTM

**Pull request merged**